src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/TextCompletionItemProvider.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+using Microsoft.VisualStudio.Editor.Razor;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+
+namespace Microsoft.CodeAnalysis.Razor.Completion
+{
+    internal class TextCompletionItemProvider : RazorCompletionItemProvider
+    {
+        private static readonly string TextTag = "text";
+        private static readonly HashSet<string> ElementCommitCharacters = new HashSet<string>{" ", ">"};
+        private readonly HtmlFactsService _htmlFactsService;
+
+        public TextCompletionItemProvider(HtmlFactsService htmlFactsService)
+        {
+            if (htmlFactsService is null)
+            {
+                throw new ArgumentNullException(nameof(htmlFactsService));
+            }
+
+            _htmlFactsService = htmlFactsService;
+        }
+
+        public override IReadOnlyList<RazorCompletionItem> GetCompletionItems(RazorSyntaxTree syntaxTree, TagHelperDocumentContext tagHelperDocumentContext, SourceSpan location)
+        {
+            var completions = new List<RazorCompletionItem>();
+            if (syntaxTree == null)
+            {
+                return completions;
+            }
+
+            var change = new SourceChange(location, string.Empty);
+            var owner = syntaxTree.Root.LocateOwner(change);
+
+            if (owner == null)
+            {
+                Debug.Fail("Owner should never be null.");
+                return completions;
+            }
+
+            if (AtTextCompletionPoint(owner))
```

Comment:
Invert to reduce nesting and aid in debuggability:
```C#
if (!AtTextCompletionPoint(owner))
{
    return Array.Empty<RazorCompletionItem>();
}
```

Nesting reduction eases in readability and the simple if-statements make debugging easier. Aka, imagine debugging over the if-statement. On a falsey result it'd jump you down an arbitrary number of lines depending on the location and when you mix in release code (it optimizes pieces out) things get even more difficult.

Reply:
Oh and having these independent if-statements helps immensely when trying to figure out why a method is doing "nothing"  with release bits 😄